### PR TITLE
adding test instructions

### PR DESCRIPTION
**Pull request by FlorianHuc**
This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] If this change is deployed to any environment (including Devnet), E2E test coverage exists or is included in this
  PR.
* [ ] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> **Low Risk**
> Low risk: adds test assets and Makefile test runners without changing core execution/prover logic; main risk is minor CI/local build friction from new targets and toolchain assumptions.
> 
> **Overview**
> Adds a small RISC-V assembly test suite under `arithmetization/src/test/examples/asm/src/` that validates key instruction behaviors (R/I/S/B/U/J types) via self-checking programs that trap on failure.
> 
> Extends `arithmetization/Makefile` with `test-instructions`, `test-xfail`, and `test-all` targets to run these guests through `zkc exec`, including an explicit *expected-failure* test (`test_xfail_jal_x0_poison.s`) to track the known `x0` hardwiring issue.
> 
> Includes a minor README tweak in the repo list section (adds/duplicates the `linea-tracer` entry).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2c79e23da3c7dbcb3da0150c01516e6795380b78. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: arithmetization/Makefile [hunk 1]
```diff
@@ -53,6 +53,62 @@ riscv-check-compilation: install-zkc zkc-compile
...
+	        fail=1; \
+	    else \
+	        echo "[XFAIL] $$t (expected)"; \
+	    fi; \
+	    rm -f /tmp/$$t.out; \
+	done; \
+	exit $$fail
+
+test-all: test-instructions test-xfail
+
```
Comment (lorenzogentile404) [inline arithmetization/Makefile:111]: This handles only the asm case, but we have tests in also `rust` and `zig` and there is another `Makefile` to handle testing: `/linea-monorepo/arithmetization/src/test/examples/Makefile`.
I would not create overlap of purpose among `Makefile`.
Also, unless there is some test case you really want to keep, I believe we are already testing enough corner cases via the `act4` framework: https://github.com/riscv/riscv-arch-test/tree/act4/tests/rv64i.

**Review by lorenzogentile404 — CHANGES_REQUESTED** [review]

File: arithmetization/Makefile (hunk 1 above)
Comment (lorenzogentile404) [reply arithmetization/Makefile:111]: Also, the comments in the tests refer to specific lines of the zkc implementation, and they may become obsolete soon when we change something.

Comment (lorenzogentile404) [reply arithmetization/Makefile:111]: If anything, I would just keep the `asm` files and do not modify the rest, but again we already have plenty of test cases.

**Comment by lorenzogentile404** [comment]
We agreed on not merging this one.
